docs/src/main/paradox/server-side/http2.md

```diff
@@ -42,13 +42,45 @@ Note that `bindAndHandle` currently does not support HTTP/2, you must use `bindA
 
 ### HTTP/2 without HTTPS
 
-While un-encrypted connections are allowed by HTTP/2, this is [generally discouraged](https://http2.github.io/faq/#does-http2-require-encryption).
+While un-encrypted connections, also known as 'h2c', are allowed by HTTP/2, this
+is [generally discouraged](https://http2.github.io/faq/#does-http2-require-encryption)
+and not typically supported by browsers.
```

Comment:
I would remove the link to say it's generally discouraged.

By default, on Kubernetes, when you use HTTP/2 or gRPC, communication is unencrypted, with ingress TLS typically being terminated at the ingress proxy. When using Istio, the sidecar terminates TLS, and communicates unencrypted locally to the application. It is possible to make the sidecar communicate using TLS with the app, but that doesn't make sense, one of the major reasons for using Istio is so that the applications don't need to worry about certificates, Istio manages it all for you. So, in these environments, which is where many people will be using gRPC, unencrypted will be the norm. It's only on the web, with browsers, where there has been a push for many years to make all communications (not just HTTP/2) encrypted, that HTTP/2 requires TLS, but that's got nothing to do with HTTP/2, and everything to do with the web and the way it's evolving.

Reply:
> So, in these environments, which is where many people will be using gRPC, unencrypted will be the norm.

That sounds reasonable in theory. Is it already happening or being prepared somewhere?

I could imagine that an HTTP/2 proxy sidecar will be its own can of worms. Is it specified which protocol it is going to use (h2c using the `Upgrade` header like in this PR or the plain form that just assumes HTTP/2 which is already implemented)?

Reply:
I haven't looked in detail enough to answer those questions.... not sure.

Reply:
I think in such situations h2c by preexisting knowledge rather than via the `Upgrade` is be more common. Makes sense to be more conservative in our assertions and remove the blanket "generally discouraged" claim though.